### DISCUSSION TITLE
Add automatic sorting by filetype to the `create new` option

### DISCUSSION THREAD

**Discussion by EllyPrice** [godotengine/godot-proposals#9463] [Editor]
A lot of us sort our files like so:

```
res://
    scenes
    scripts
    images
    audio
    etc...
```

So wouldn't it be cool if there was an option somewhere in the editor to say "Hey, whenever I make a new .(filetype) put it into (selected folder)!" That way whenever a new scene or script or whatever is created, it automatically goes to whatever folder you've assigned. 
This was kind of asked already [here](https://godotengine.org/qa/83852/sort-godot-filesystem-by-filetype), but my proposal negates ever needing to resort in the first place. 

### Reasons why this should be implemented: 
1. Moving a file's location can often cause catastrophic issues (one time doing this back when I still used Windows, and before utilizing version control, it corrupted an entire project).
2. Quality of life. If most of us do this anyways, why not just add an option you can check that does it automatically for you on creation of the script?

Often what I'll find myself doing is having really good programming patterns early on, ensuring I put every script in the right folder before I even start editing them, but when I'm several days into a project that goes out the window due to being unable to switch focus. If something just did it automatically, not only would it help me but I'm sure plenty of others.

### How this should be implemented (in my opinion, anyway):
- Create an option in Project Settings to add an "auto-folder" (folder that automatically takes a certain filetype) with a input field to give it a name
- After selecting "create" it gives the user the option to either type or select an extension (.gd, .tscn, .json, .png, etc.)
- User clicks "OK" and then a new, unique looking folder (either make it a different icon or color than the normal folders) appears under res:// with whatever name the user typed
- When the user creates a new filetype (e.g. when you attach a script to a Node object), if the filetype matches the one in the Project Settings then rather than that file being added to whatever folder the user had open (what happens currently) it is added to one of their premade auto-folders with a matching filetype filter

I'm still rusty at C++ but I'd like to help if this gets picked up.

**Comment by SB1O1** [comment]
Would be so usefull

**Reply by Calinou** [reply]
@SB1O1 Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.
